### PR TITLE
fix: function-expression as object value.

### LINKED ISSUE #723: Cannot read property '0' of undefined
When trying to generate the below schema, `ts-json-schema-generator` (v0.89.0) fails with `TypeError: Cannot read property '0' of undefined`.

```typescript
const makeDefaultItem = () => ({vendorable: false});

export const PaintJobItems = {
  paintJobBrush: makeDefaultItem(),
};

export type PaintJobItemsType = typeof PaintJobItems;
```

Removing the method and including the type directly (below) doesn't generate the error and generates a valid schema:
```typescript
export const PaintJobItems = {
  paintJobBrush: {vendorable: false},
};

export type PaintJobItemsType = typeof PaintJobItems;
```

This seems to be an issue with `CallExpressionParser` here: https://github.com/vega/ts-json-schema-generator/blob/master/src/NodeParser/CallExpressionParser.ts#L19. Changing that line to:
```typescript
if ("typeArguments" in type && type.typeArguments) {
```
fixes the issue.

### PR DESCRIPTION
**Pull request by acornwall**
Fixes #723.

### CODE REVIEW

File: test/valid-data-struct.test.ts [hunk 1]
```diff
@@ -4,6 +4,7 @@ describe("valid-data-struct", () => {
     // TODO: generics recursive
     it("simple-object", assertValidSchema("simple-object", "SimpleObject"));
 
+    it.only("object-function-expression", assertValidSchema("object-function-expression", "MyType"));
```
Comment (domoritz) [inline test/valid-data-struct.test.ts:7]: Remove the only

**Review by domoritz — CHANGES_REQUESTED** [review]

File: test/valid-data-struct.test.ts (hunk 1 above)
Comment (acornwall) [reply test/valid-data-struct.test.ts:7]: Omg, can't believe I forgot to remove that. Thanks for catching this.

File: src/NodeParser/CallExpressionParser.ts [hunk 2]
```diff
@@ -16,7 +16,7 @@ export class CallExpressionParser implements SubNodeParser {
         const type = this.typeChecker.getTypeAtLocation(node);
 
         // FIXME: remove special case
-        if ("typeArguments" in type) {
+        if ("typeArguments" in type && (type as any).typeArguments) {
```
Comment (domoritz) [inline src/NodeParser/CallExpressionParser.ts:19]: Could this be just 

```suggestion
        if ((type as any)?.typeArguments) {
```

**Review by domoritz — APPROVED** [review]
Thank you for the pull request. I have one small idea but looks good otherwise.

**Comment by domoritz** [comment]
Thank you @acornwall!

**Pull request merged**